bot.go

```diff
@@ -401,28 +400,37 @@ func (bot *BotAPI) GetUpdatesChan(config UpdateConfig) UpdatesChannel {
 			for _, update := range updates {
 				if update.UpdateID >= config.Offset {
 					config.Offset = update.UpdateID + 1
-					ch <- update
+
+					select {
```

Comment:
That's not good. This place make execution nondeterministic, because cases will be chosen randomly, if both channels are ready. Also we must be sure we won't process same update twice (before app close and after restart). I think as a start we can add

```
select {
	case <-done:
		return
	default:
}
```
on line 399. We will get updates and ignore them all at once. After restart telegram API will return them again, because they were unconfirmed.
As a next step, in another PR, we can add `force` flag to `Shutdown()` method to skip updates in a loop (line 400) even after we started to process them.